src/charm.py

```diff
@@ -254,6 +255,38 @@ def _restart(self, event: EventBase) -> None:
 
         self.snap.restart_snap_service("kafka")
 
+        if broker_active(
```

Comment:
Can this be extracted into a separate method? as the logic seems to be the same as in `_disable_enable_restart`